### DISCUSSION TITLE
Best way to spawn individual machines dynamically

### DISCUSSION THREAD

**Discussion by nickngqs** [statelyai/xstate#3838] [Q&A]
What would be the best way to dynamically create machines for an approval workflow.

Scenario
Person A submits request for approval. Person A is able to select any number of Approvers required.

The workflow machine will generate individual Approve/Reject machine for each of the approvers.

When all individual machines is approved, the entire request is considered Approved.
When any individual rejects, the entire request is considered rejected.

What will be the best way to allow this kind of states to be visualised in stately?

![Slice 1](https://user-images.githubusercontent.com/3901903/219249088-eb3f99c6-a423-4f2b-bcb8-c57cbea5da5d.png)

**Comment by davidkpiano** [comment]
Since it's a dynamic number, you can either use `spawn(...)` to spawn an individual approval workflow for every person and send back events to the parent upon approval/rejection.

Then, you can use `always: ...` to check that all of the approvals have been made.

On a rejection event, you can send an event to each actor to cancel those approval workflows and transition to a state where a rejection happened.

**Reply by nickngqs** [reply]
thanks that's helpful!
